pkg/keycloak/realm/phaseHandler.go

```diff
@@ -247,6 +246,77 @@ func (ph *phaseHandler) reconcileUser(kcUser, specUser *v1alpha1.KeycloakUser, r
 		}
 	}
 
+	if err := ph.reconcileUserClientRoles(specUser, realmName, authenticatedClient); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (ph *phaseHandler) reconcileUserClientRoles(specUser *v1alpha1.KeycloakUser, realmName string, authenticatedClient keycloak.KeycloakInterface) error {
+	clients, err := authenticatedClient.ListClients(realmName)
+	if err != nil {
+		return err
+	}
+	for clientName, roles := range specUser.ClientRoles {
+		rolesCopy := make([]string, len(roles))
+		copy(rolesCopy, roles)
+	FindMatchingClient:
+		for _, client := range clients {
+			if clientName == client.ClientID {
+				err = ph.reconcileRolesForClient(rolesCopy, client, specUser, realmName, authenticatedClient)
```

Comment:
ignoring err here we should at least log it and perhaps gather a multi error?

Reply:
good spot, fixed now